overlord/snapstate/handlers.go

```diff
@@ -997,7 +1036,73 @@ func (m *SnapManager) undoRefreshAliasesV2(t *state.Task, _ *tomb.Tomb) error {
 
 	snapst.AutoAliasesDisabled = autoDisabled
 	snapst.Aliases = oldAliases
-	Set(st, snapName, snapst)
+	newSnapStates := make(map[string]*SnapState, 1+len(otherSnapDisabled))
+	newSnapStates[snapName] = snapst
+
+	// if we disabled other snap aliases try to undo that
+	var conflicting map[string]bool
+	var curOtherSnapStates map[string]*SnapState
+	if len(otherSnapDisabled) != 0 {
```

Comment:
We might deindent the whole block and let the zero case of these statements to their job. The cost will be pretty close to zero, and will increase readability just a tiny bit.

Not a requirement, obviously.